pkg/controller/gittrack/gittrack_controller_test.go

```diff
@@ -48,1075 +49,1414 @@ import (
-				setsMetric("ignored", 0.0)
-				setsMetric("inSync", 0.0)
+				It("sends events about creating GitTrackObjects", func() {
+					events := &v1.EventList{}
+					Eventually(func() error { return c.List(context.TODO(), events) }, timeout).Should(Succeed())
+					startEvents := testevents.Select(events.Items, reasonFilter("CreateStarted"))
+					successEvents := testevents.Select(events.Items, reasonFilter("CreateSuccessful"))
+					Expect(startEvents).ToNot(BeEmpty())
+					Expect(successEvents).ToNot(BeEmpty())
+					for _, e := range append(startEvents, successEvents...) {
+						Expect(e.InvolvedObject.Kind).To(Equal("GitTrack"))
+						Expect(e.InvolvedObject.Name).To(Equal("example"))
+						Expect(e.Type).To(Equal(string(v1.EventTypeNormal)))
+					}
+				})
 			})
 
-			It("creates GitTrackObjects", func() {
-				deployGto := &farosv1alpha1.GitTrackObject{}
-				serviceGto := &farosv1alpha1.GitTrackObject{}
-				Eventually(func() error {
-					return c.Get(context.TODO(), types.NamespacedName{Name: "deployment-nginx", Namespace: "default"}, deployGto)
-				}, timeout).Should(Succeed())
-				Eventually(func() error {
-					return c.Get(context.TODO(), types.NamespacedName{Name: "service-nginx", Namespace: "default"}, serviceGto)
-				}, timeout).Should(Succeed())
+			Context("with multi-document YAML", func() {
+				BeforeEach(func() {
+					createInstance(instance, "9bf412f0e893c8c1624bb1c523cfeca8243534bc")
+					// Wait for client cache to expire
+					waitForInstanceCreated(key)
+				})
+
+				It("creates GitTrackObjects", func() {
+					dsGto, cmGto := &farosv1alpha1.GitTrackObject{}, &farosv1alpha1.GitTrackObject{}
+					Eventually(func() error {
+						return c.Get(context.TODO(), types.NamespacedName{Name: "daemonset-fluentd", Namespace: "default"}, dsGto)
+					}, timeout).Should(Succeed())
+					Eventually(func() error {
+						return c.Get(context.TODO(), types.NamespacedName{Name: "configmap-fluentd-config", Namespace: "default"}, cmGto)
+					}, timeout).Should(Succeed())
+				})
 			})
 
-			It("sets ownerReferences for created GitTrackObjects", func() {
-				deployGto := &farosv1alpha1.GitTrackObject{}
-				serviceGto := &farosv1alpha1.GitTrackObject{}
-				Eventually(func() error {
-					return c.Get(context.TODO(), types.NamespacedName{Name: "deployment-nginx", Namespace: "default"}, deployGto)
-				}, timeout).Should(Succeed())
-				Eventually(func() error {
-					return c.Get(context.TODO(), types.NamespacedName{Name: "service-nginx", Namespace: "default"}, serviceGto)
-				}, timeout).Should(Succeed())
-				Expect(len(deployGto.OwnerReferences)).To(Equal(1))
-				Expect(len(serviceGto.OwnerReferences)).To(Equal(1))
+			Context("with a cluster scoped resource", func() {
+				BeforeEach(func() {
+					createInstance(instance, "b17c0e0f45beca3f1c1e62a7f49fecb738c60d42")
+					// Wait for client cache to expire
+					waitForInstanceCreated(key)
+				})
+
+				It("does not create ClusterGitTrackObject", func() {
```

Comment:
I think the test here needs updating, you've updated the description but the test case does not make sense with the current description